azure-iot-hub/tests/test_iothub_amqp_client.py

```diff
@@ -0,0 +1,72 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import pytest
+import time
+import base64
+import hmac
+import hashlib
+import copy
+import logging
+import uamqp
+from azure.iot.hub.iothub_amqp_client import IoTHubAmqpClient
+
+try:
+    from urllib import quote, quote_plus, urlencode  # Py2
+except Exception:
+    from urllib.parse import quote, quote_plus, urlencode
+
```

Comment:
nit: this is why we have the six library.  preferred style is to say "import six.moves.urllib as urllib" then access these as "urllib.parse.quote" in the code.  This way you don't have to try/except to get py2/py3 compatibility.